app/src/main/java/fr/free/nrw/commons/utils/DialogUtil.kt

```diff
@@ -104,65 +95,76 @@ object DialogUtil {
         )
     }
 
-    /*
-    Shows alert dialog with custom view
-     */
-    private fun showAlertDialog(
+    @JvmStatic
+    fun showAlertDialog(
         activity: Activity,
         title: String,
         message: String,
-        positiveButtonText: String,
-        negativeButtonText: String,
+        positiveButtonText: String?,
         onPositiveBtnClick: Runnable?,
-        onNegativeBtnClick: Runnable?,
-        customView: View?,
-        cancelable: Boolean
+        cancellable: Boolean
     ) {
-        // If the custom view already has a parent, there is already a dialog showing with the view
-        // This happens for on resume - return to avoid creating a second dialog - the first one
-        // will still show
-        if (customView?.parent != null) {
-            return
-        }
-
-        showSafely(activity, AlertDialog.Builder(activity).apply {
-            setTitle(title)
-            setMessage(message)
-            setView(customView)
-            setCancelable(cancelable)
-            setPositiveButton(positiveButtonText) { _: DialogInterface, _: Int ->
-                onPositiveBtnClick?.run()
-            }
-            setNegativeButton(negativeButtonText) { _: DialogInterface, _: Int ->
-                onNegativeBtnClick?.run()
-            }
-        }.create())
+        showAlertDialog(
+            activity,
+            title,
+            message,
+            positiveButtonText,
+            onPositiveBtnClick,
+            cancellable
+        )
     }
 
     /**
-     * show a dialog with just a positive button
+     * show a dialog
      * @param activity
      * @param title
      * @param message
      * @param positiveButtonText
-     * @param positiveButtonClick
-     * @param cancellable
+     * @param negativeButtonText
+     * @param onPositiveBtnClick
+     * @param onNegativeBtnClick
+     * @param customView
+     * @param cancelable
      */
     @JvmStatic
     fun showAlertDialog(
         activity: Activity,
         title: String,
         message: String,
-        positiveButtonText: String,
-        positiveButtonClick: Runnable?,
-        cancellable: Boolean
+        positiveButtonText: String? = let { "" },
```

Comment:
should be `String? = null` because null is an okay default, this may not have been set by a caller.

This let usage is quite confusing, this doesn't compile, right?

Reply:
I am not familiar with **let** in Kotlin 😅 It did actually compiled for me but I have made the changes!

Reply:
@macgills I don't know if this is the right thing to do here
```
positiveButtonText?.let {
                if (positiveButtonText.isBlank()) {
                    setPositiveButton(positiveButtonText) { _: DialogInterface, _: Int ->
                        onPositiveBtnClick?.run()
                    }
                }
            }
```